app/assets/src/components/common/CoverageVizSidebar/CoverageVizSidebar.jsx

```diff
@@ -0,0 +1,367 @@
+import React from "react";
+import { min, max, find, get } from "lodash/fp";
+import cx from "classnames";
+
+import Sidebar from "~/components/ui/containers/Sidebar";
+import PropTypes from "~/components/utils/propTypes";
+import Histogram from "~/components/visualizations/Histogram";
+import GenomeViz from "~/components/visualizations/GenomeViz";
+import Dropdown from "~ui/controls/dropdowns/Dropdown";
+import LoadingIcon from "~ui/icons/LoadingIcon";
+import BasicPopup from "~/components/BasicPopup";
+import cs from "./coverage_viz_sidebar.scss";
+
+// Temporary data.
+// TODO(mark): Fetch data from the server.
+import CUTI_ACCESSION_ONE from "./temporary_data/CP012352.1_coverage_viz.json";
+import CUTI_ACCESSION_TWO from "./temporary_data/CP012647.1_coverage_viz.json";
+import CUTI_ACCESSION_THREE from "./temporary_data/AE017283.1_coverage_viz.json";
+import CUTI_ACCESSION_FOUR from "./temporary_data/CP012350.1_coverage_viz.json";
+
+const ACCESSION_COVERAGE_DATA = {
+  "CP012352.1": CUTI_ACCESSION_ONE,
+  "CP012647.1": CUTI_ACCESSION_TWO,
+  "AE017283.1": CUTI_ACCESSION_THREE,
+  "CP012350.1": CUTI_ACCESSION_FOUR
+};
+
+const METRIC_TOOLTIPS = {
+  referenceAccession: "Genbank Accession ID for the reference accession.",
+  referenceLength: "Length in bp of the reference accession sequence.",
+  alignedContigs:
+    "Number of contigs for which this accession was the best match.",
+  avgAlignedLength:
+    "Length of aligned regions, averaged over all reads and contigs ",
+  avgCoverageDepth:
+    "The total length of all aligned contigs and reads, divided by the accession length.",
+  coverageBreadth:
+    "The percentage of the accession that is covered by at least one read or contig.",
+  alignedReads: "Number of reads for which this accession was the best match.",
+  avgMismatchedPercent:
+    "Percentage of aligned regions that are mismatches, averaged over all reads and contigs."
+};
+
+const METRIC_COLUMNS = [
+  [
+    {
+      key: "referenceAccession",
+      name: "Reference Accession"
+    },
+    {
+      key: "referenceLength",
+      name: "Reference Length"
+    }
+  ],
+  [
+    {
+      key: "alignedContigs",
+      name: "Aligned Contigs"
+    },
+    {
+      key: "avgAlignedLength",
+      name: "Avg. Aligned Length"
+    }
+  ],
+  [
+    {
+      key: "avgCoverageDepth",
+      name: "Avg. Coverage Depth"
+    },
+    {
+      key: "coverageBreadth",
+      name: "Coverage Breadth"
+    }
+  ],
+  [
+    {
+      key: "alignedReads",
+      name: "Aligned Reads"
+    },
+    {
+      key: "avgMismatchedPercent",
+      name: "Avg. Mismatched %"
+    }
+  ]
+];
+
+const alignRange = range => [min(range), max(range)];
+
+export default class CoverageVizSidebar extends React.Component {
+  _accessionDataCache = {};
+
+  state = {
+    currentAccession: null
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevProps.visible && this.props.visible) {
```

Comment:
Can we use the pattern of loading all `props` and `state` variables in the beginning? 
i.e `const { visible, params } = this.props`

Reply:
Done